files/brews/nginx.rb

```diff
@@ -1,67 +1,206 @@
   end
 
   def install
+    # Changes default port to 8080
+    inreplace "conf/nginx.conf" do |s|
+      s.gsub! "listen       80;", "listen       8080;"
+      s.gsub! "    #}\n\n}", "    #}\n    include servers/*;\n}"
+    end
+
     pcre = Formula["pcre"]
-    openssl = Formula["openssl"]
-    cc_opt = "-I#{pcre.include} -I#{openssl.include}"
-    ld_opt = "-L#{pcre.lib} -L#{openssl.lib}"
-
-    args = ["--prefix=#{prefix}",
-            "--with-http_ssl_module",
-            "--with-pcre",
-            "--with-ipv6",
-            "--with-cc-opt=#{cc_opt}",
-            "--with-ld-opt=#{ld_opt}",
-            "--conf-path=/opt/boxen/config/nginx/nginx.conf",
-            "--pid-path=/opt/boxen/data/nginx/nginx.pid",
-            "--lock-path=/opt/boxen/data/nginx/nginx.lock"]
-
-    args << passenger_config_args if ARGV.include? '--with-passenger'
-    args << "--with-http_dav_module" if ARGV.include? '--with-webdav'
-    args << "--with-http_gzip_static_module" if ARGV.include? '--with-gzip-static'
-    args << "--with-http_v2_module" if ARGV.include? "--with-http2"
-
-    system "./configure", *args
-    system "make"
-    system "make install"
-    man8.install "objs/nginx.8"
-
-    # remove unnecessary config files
-    system "rm -rf #{etc}/nginx"
+
+    if build.with? "passenger"
+      openssl = Formula["openssl"]
+    else
+      openssl = Formula["openssl@1.1"]
+    end
+
+    cc_opt = "-I#{pcre.opt_include} -I#{openssl.opt_include}"
+    ld_opt = "-L#{pcre.opt_lib} -L#{openssl.opt_lib}"
+
+    args = %W[
+      --prefix=#{prefix}
+      --with-http_ssl_module
+      --with-pcre
+      --sbin-path=#{bin}/nginx
+      --with-cc-opt=#{cc_opt}
+      --with-ld-opt=#{ld_opt}
+      --conf-path=#{etc}/nginx/nginx.conf
+      --pid-path=#{var}/run/nginx.pid
+      --lock-path=#{var}/run/nginx.lock
+      --http-client-body-temp-path=#{var}/run/nginx/client_body_temp
+      --http-proxy-temp-path=#{var}/run/nginx/proxy_temp
+      --http-fastcgi-temp-path=#{var}/run/nginx/fastcgi_temp
+      --http-uwsgi-temp-path=#{var}/run/nginx/uwsgi_temp
+      --http-scgi-temp-path=#{var}/run/nginx/scgi_temp
+      --http-log-path=#{var}/log/nginx/access.log
+      --error-log-path=#{var}/log/nginx/error.log
+      --with-http_gzip_static_module
+      --with-http_v2_module
+    ]
+
+    if build.with? "passenger"
+      nginx_ext = `#{Formula["passenger"].opt_bin}/passenger-config --nginx-addon-dir`.chomp
+      args << "--add-module=#{nginx_ext}"
+    end
+
+    args << "--with-http_dav_module" if build.with? "webdav"
+    args << "--with-debug" if build.with? "debug"
+    args << "--with-http_gunzip_module" if build.with? "gunzip"
+
+    if build.head?
+      system "./auto/configure", *args
+    else
+      system "./configure", *args
+    end
+
+    system "make", "install"
+    if build.head?
+      man8.install "docs/man/nginx.8"
+    else
+      man8.install "man/nginx.8"
+    end
+  end
+
+  def post_install
+    (etc/"nginx/servers").mkpath
```

Comment:
Where "this" is everything inside of the `post_install`. Apologies for the ambiguity.